src/main/java/io/openmanufacturing/ame/resolver/inmemory/InMemoryStrategy.java

```diff
@@ -58,54 +71,95 @@ public Try<Model> apply( final AspectModelUrn aspectModelUrn ) {
+      if ( namedResourceFile.exists() ) {
+         return loadFromUri( namedResourceFile.toURI() );
+      }
 
-      final String aspectModelUrn = sdsStatements.orElseThrow(
-            () -> new NotImplementedError( "AspectModelUrn cannot be found." ) ).next().getSubject().getURI();
+      LOG.warn( "Looking for {}, but no {}.ttl was found. Inspecting files in {}", aspectModelUrn.getName(),
+            aspectModelUrn.getName(), directory );
 
-      return AspectModelUrn.fromUrn( aspectModelUrn );
+      return Arrays.stream( Optional.ofNullable( directory.toFile().listFiles() ).orElse( new File[] {} ) )
+                   .filter( File::isFile ).filter( file -> file.getName().endsWith( ".ttl" ) ).map( File::toURI )
+                   .sorted().map( this::loadFromUri ).filter(
+                  tryModel -> tryModel.map( model -> AspectModelResolver.containsDefinition( model, aspectModelUrn ) )
+                                      .getOrElse( false ) ).findFirst().orElse( Try.failure( new FileNotFoundException(
+                  "No model file containing " + aspectModelUrn + " could be found in directory: " + directory ) ) );
+   }
+
+   public AspectModelUrn getAspectModelUrn() {
+      return AspectModelUrn.fromUrn(
+            getSdsStatements().orElseThrow( () -> new NotImplementedError( "AspectModelUrn cannot be found." ) ).next()
+                              .getSubject().getURI() );
    }
 
    private Optional<StmtIterator> getSdsStatements() {
+      final List<StmtIterator> stmtIterators = new ArrayList<>();
 
-      for ( final KnownVersion version : KnownVersion.getVersions() ) {
-         final BAMM bamm = new BAMM( version );
-         final List<Resource> resources = List.of( bamm.Aspect(), bamm.Property(), bamm.Entity(), bamm.Characteristic(),
-               bamm.Constraint() );
-         final List<StmtIterator> collect = resources.stream().filter(
-                                                           resource -> model.listStatements( null, RDF.type, resource ).hasNext() )
-                                                     .map( resource -> model.listStatements( null, RDF.type,
-                                                           resource ) ).toList();
-         if ( !collect.isEmpty() ) {
-            return collect.stream().findFirst();
-         }
-      }
+      KnownVersion.getVersions()
+                  .forEach( version -> stmtIterators.addAll( getListOfAllBAMMElements( version )
+                        .stream()
+                        .filter( resource -> aspectModel.listStatements( null, RDF.type, resource ).hasNext() )
+                        .map( resource -> aspectModel.listStatements( null, RDF.type, resource ) )
+                        .toList() ) );
+
+      return stmtIterators.isEmpty() ? Optional.empty() : stmtIterators.stream().findFirst();
+   }
 
-      return Optional.empty();
+   private List<Resource> getListOfAllBAMMElements( final KnownVersion version ) {
```

Comment:
The method does not return _elements_ but _types of elements_, the name should be mor expressive. It seems to me this should be a functionality of the BAMM vocabulary itself (or even of the meta model package). Maybe create an issue for that?
Also: How about bamm-e elements?

Reply:
Yeah good idea. I will add an issue for that.